src/protocol/handshakestate.c

```diff
@@ -52,13 +52,12 @@
  *
  * \param flags The flags from the handshake pattern.
  * \param prefix_id The prefix identifier from the protocol name.
- * \param role The role, either initiator or responder.
  * \param is_fallback Non-zero if we are processing a fallback pattern.
  *
  * \return The key requirements for the handshake pattern.
  */
 static int noise_handshakestate_requirements
-    (NoisePatternFlags_t flags, int prefix_id, int role, int is_fallback)
+    (NoisePatternFlags_t flags, int prefix_id, int is_fallback)
```

Comment:
This interface must not be changed like that. I prefer that you use the `__attribute_((unused))` rather than removing that

Reply:
You can also write `static int noise_handshakestate_requirements(NoisePatternFlags_t flags, int prefix_id, int, int is_fallback)` and for the compiler it's an unused parameter